packages/early-boot-config/early-boot-config.spec

```
@@ -0,0 +1,163 @@
+%global _cross_first_party 1
+%undefine _debugsource_packages
+
+Name: %{_cross_os}early-boot-config
+Version: 0.0
+Release: 0%{?dist}
+Summary: early-boot-config
+License: Apache-2.0 OR MIT
+URL: https://github.com/bottlerocket-os/bottlerocket
+
+Source100: early-boot-config.service
+
+BuildRequires: %{_cross_os}glibc-devel
+
+%description
+%{summary}.
+
+%package -n %{_cross_os}early-boot-config-local
+Summary: local-provider
+
+%description -n %{_cross_os}early-boot-config-local
+%{summary}.
+
+%package -n %{_cross_os}early-boot-config-aws
+Summary: early-boot-config package for AWS
+Requires: %{name}
+Requires: %{_cross_os}early-boot-config-local
+
+%description -n %{_cross_os}early-boot-config-aws
+%{summary}.
+
+%ifarch x86_64
+%package -n %{_cross_os}early-boot-config-vmware
+Summary: early-boot-config package for vmware
+Requires: %{name}
+Requires: %{_cross_os}early-boot-config-local
+
+%description -n %{_cross_os}early-boot-config-vmware
+%{summary}.
+%endif
+
+%package -n %{_cross_os}early-boot-config-metal
+Summary: early-boot-config package for metal
+Requires: %{name}
+Requires: %{_cross_os}early-boot-config-local
+
+%description -n %{_cross_os}early-boot-config-metal
+%{summary}.
+
+%prep
+%setup -T -c
+%cargo_prep
+
+%build
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p early-boot-config
+
+# build aws user data providers
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p ec2-identity-doc-user-data-provider \
+    -p ec2-imds-user-data-provider
+
+# build local file user data providers
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p local-defaults-user-data-provider \
+    -p local-file-user-data-provider \
+    -p local-overrides-user-data-provider
+
+%ifarch x86_64
+# build vmware user data providers
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p vmware-cd-rom-user-data-provider \
+    -p vmware-guestinfo-user-data-provider
+%endif
```

I'd expect the build to be faster if we run them all in parallel -
```suggestion
%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
    -p early-boot-config \
    -p ec2-identity-doc-user-data-provider \
    -p ec2-imds-user-data-provider \
    -p local-defaults-user-data-provider \
    -p local-file-user-data-provider \
    -p local-overrides-user-data-provider \
%ifarch x86_64
    -p vmware-cd-rom-user-data-provider \
    -p vmware-guestinfo-user-data-provider \
%endif
  %{nil}
```